implementation/src/main/java/io/smallrye/mutiny/infrastructure/Infrastructure.java

```diff
@@ -73,6 +73,9 @@ public static void setDefaultExecutor(Executor s) {
         if (existing instanceof ExecutorService) {
             ((ExecutorService) existing).shutdownNow();
         }
+        if (DEFAULT_SCHEDULER != null) {
+            DEFAULT_SCHEDULER.shutdownNow();
```

Comment:
As in general the scheduler is back up by the executor, isn't shutting down the executor enough

Reply:
It's also my thought, but I thought I'd make sure the wrapper would also be shutdown. Perhaps there's some transient state.

I've have multiple good runs with that PR, but I'm not 100% sure it won't pop up again.

Reply:
I also tried to repeat the flaky test on my machine 10k times, no issue.